src/main/java/com/intel/gkl/compression/IntelDeflaterFactory.java

```diff
@@ -31,7 +31,7 @@ public IntelDeflaterFactory() {
      */
     public Deflater makeDeflater(final int compressionLevel, final boolean gzipCompatible) {
         if (intelDeflaterSupported) {
-            if ((compressionLevel == 1 && gzipCompatible) || compressionLevel != 1) {
+            if (compressionLevel != 1 || gzipCompatible) {
```

Comment:
Did we try to run level =1 and gzip?

Reply:
Yes, it works ok.

Reply:
If the level is set to 1 and gzip set to true, the second condition is satisfied, so the `makeDeflater` function returns IntelDeflater object